packages/training/python/tests/test_enhanced_rewards_extended.py

```
@@ -0,0 +1,1130 @@
+    
+    def test_zero_starting_balance(self):
+        """Zero starting balance returns 0 reward."""
+        reward = regime_adjusted_pnl_reward(
+            actual_pnl=1000,
+            starting_balance=0,
+            regime_overall="bull",
+            regime_volatility=0.5,
+            regime_expected_return=0.05,
+        )
+        # Should handle gracefully
+        assert reward == 0.0 or math.isnan(reward) == False
```

Ruff (0.14.11)

Avoid equality comparisons to `False`; use `not math.isnan(reward):` for false checks

Replace with `not math.isnan(reward)`

(E712)